doc/lua_api.txt

```diff
@@ -414,6 +414,100 @@ the word "`alpha`", then each texture pixel will contain the RGB of
+The color channels (RGB) of resulting pixel are calculated using this formula:
+
+	red = source_red * source_factor + destination_red * destination_factor
+
+
+Some examples:
+
+Disabled blend (default):
+
+	blend_type = minetest.blend_type.none,
+
+Use this value to disable blending. The texture will be applied to existing pixels
+using their alpha channel. It's recomended to use 1-bit alpha.
+
+Additive blend:
+
+	blend_type = minetest.blend_type.additive,
+
+source_factor is taken from the textures alpha channel,
+destination_factor is fixed at 1.
+For example with a source color (0,192,255,63) and already rendered
+pixel color (40,192,0) the calculation looks like this:
+
+R = 0 (source_red) * 63/255 (source_factor) +
+    40 (destination_red) * 1 (destination_factor)
+  = 0 * 63/255 + 40 * 1 = 40
+G = 192 * 63/255 + 192 * 1 = 239
+B = 255 * 63/255 + 0 * 1 = 63
+
+Additive blend can e.g. be used to make nice-looking fire particles
+using the color (102,34,17) which will change color from red to yellow the more
+particles are stacked ontop of eachother.
+
+Subtractive blend:
+	
+	blend_type = minetest.blend_type.subtractive,
+
+source_factor is fixed at 0, destination_factor is taken from the texture.
+Texture darkness acts like an alpha channel.
+
+Invert blend:
+
+	blend_type = minetest.blend_type.invert,
+
+source_factor is the inverted color of existing pixels, destination_factor is
+the inverted color of the texture.
+Resulting pixels have inverted color if the source color value is big enough,
+otherwise they appear black.
+
+
+It is possible combine the primitives below to design a custom blending mode
+'minetest.pack_texture_blend_func' will take take these and return an integer
+to be used as 'blend_type' in particle definitions.
+
+A values in a brackets specify multiplicators of red, green, blue
+and alpha channels respectively.
+
+* Blend types (for srcFact & dstFact):
+    *  "zero"                -- src & dest (0, 0, 0, 0)
+    *  "one"                 -- src & dest (1, 1, 1, 1)
+    *  "dst_color"           -- src (destR, destG, destB, destA)
+    *  "dst_color_inv"       -- src (1-destR, 1-destG, 1-destB, 1-destA)
+    *  "src_color"           -- dest (srcR, srcG, srcB, srcA)
+    *  "src_color_inv"       -- dest (1-srcR, 1-srcG, 1-srcB, 1-srcA)
+    *  "src_alpha"           -- src & dest (srcA, srcA, srcA, srcA)
+    *  "src_alpha_inv"       -- src & dest (1-srcA, 1-srcA, 1-srcA, 1-srcA)
+    *  "dst_alpha"           -- src & dest (destA, destA, destA, destA)
+    *  "dst_alpha_inv"       -- src & dest (1-destA, 1-destA, 1-destA, 1-destA)
+    *  "src_alpha_saturate"  -- src (min(srcA, 1-destA), ...)
+
+* Modulate values (for modulate):
+    *  "1x" -- multiply components by 1 (no shift)
+    *  "2x" -- multiply components by 2 (x << 1)
+    *  "4x" -- multiply components by 4 (x << 2)
+
+* Alpha sources (for alphaSource):
+    * "none"         -- no alpha
+    * "vertex"       -- use alpha from existing pixel
```

Comment:
Vertex - use alpha from vertex colour, i.e. you can specify alpha value when you define vertex. Because in Minetest particles currently vertex alpha always equals 255 setting this will do nothing.